packages/web3-providers/src/rarible/index.ts

```diff
@@ -135,9 +167,193 @@ export async function getHistoryFromRarible(tokenAddress: string, tokenId: strin
+                  profile_img_url: '',
+                  user: { username: owner },
+                  link: '',
+              }
+            : null,
+        creator: creator
+            ? {
+                  address: creator.account,
+                  profile_img_url: '',
+                  user: { username: creator.account },
+                  link: '',
+              }
+            : null,
+        traits: asset?.meta.attributes.map(({ key, value }) => ({ trait_type: key, value })),
+        description: asset?.meta.description ?? '',
+        name: asset?.meta.name ?? 'Unknown',
+        collection_name: '',
+        animation_url: asset.meta.animation?.url.PREVIEW,
+        current_price: 0,
+        current_symbol: 'ETH',
+        end_time: null,
+        order_payment_tokens: [] as FungibleTokenDetailed[],
+        offer_payment_tokens: [] as FungibleTokenDetailed[],
+        top_ownerships: owner
+            ? [
+                  {
+                      owner: {
+                          address: owner,
+                          profile_img_url: '',
+                          user: { username: owner },
+                          link: '',
+                      },
+                  },
+              ]
+            : null,
+        slug: '',
+        response_: asset,
+    } as NFTAsset
+}
+
+async function _getAsset(address: string, tokenId: string) {
+    const assetResponse = await fetchFromRarible<RaribleNFTItemMapResponse>(
+        RaribleChainURL,
+        urlcat('/v0.1/nft/items/:address::tokenId', {
+            includeMeta: true,
+            address,
+            tokenId,
+        }),
+        {
+            method: 'GET',
+            mode: 'cors',
+            headers: {
+                'content-type': 'application/json',
+            },
+        },
+    )
+    return assetResponse
+}
+
+export async function getAsset(address: string, tokenId: string, chainId: ChainId) {
+    const asset = await _getAsset(address, tokenId)
+    if (!asset) return
+
+    return createNFTAsset(asset, chainId)
+}
+
+export async function getNFT(tokenAddress: string, tokenId: string) {
+    const asset = await _getAsset(tokenAddress, tokenId)
+    return createERC721TokenAsset(tokenAddress, tokenId, asset)
+}
+
+export async function getNFTs(from: string, chainId: ChainId) {
+    const params = new URLSearchParams()
+    params.append('owner', from)
+    const assetResponse = await fetchFromRarible<{ total: number; items: RaribleNFTItemMapResponse[] }>(
+        RaribleChainURL,
+        `/v0.1/nft/items/byOwner?${params.toString()}`,
+        {
+            method: 'GET',
+            mode: 'cors',
+            headers: {
+                'content-type': 'application/json',
+            },
+        },
+    )
+    return assetResponse.items.map((asset) => createERC721TokenAsset(asset.contract, asset.tokenId, asset))
+}
+
+export async function getContractBalance(from: string, contract_address: string, chainId: ChainId) {
```

Comment:
We prefer the camel case, the snack_style only for API or something else.

```suggestion
export async function getContractBalance(from: string, contractAddress: string, chainId: ChainId) {
```